### PR TITLE
Truncate long bundle names to fix #47

### LINKED ISSUE #47: Generated file names for bundles can exceed file system limit
I tried updating a large project to test the latest Steal and discovered the build won't complete because the filename generated for a widely shared bundle is too long for the file system. 

This is the same issue that I proposed a pull-request on in legacy StealJS in https://github.com/bitovi/steal/pull/170 - NOTE: GitHub seems to have mangled that page with history from a separate issue. The relevant comments are [Justin's](https://github.com/justinbmeyer) and [my response](https://github.com/bitovi/steal/pull/170#issuecomment-26483421).

In that PR I suggest taking truncating long file names and appending a hash, which I think is still reasonable.

### PR DESCRIPTION
**Pull request by asavoy**
Proposed fix for #47.

### CODE REVIEW

File: test/long_bundle_names/stealconfig.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+		"app_b_with_a_very_long_name_abcdefghijklmnopqrstuvwxyz_0123456789/app_b_with_a_very_long_name_abcdefghijklmnopqrstuvwxyz_0123456789" : "app_b_with_a_very_long_name_abcdefghijklmnopqrstuvwxyz_0123456789",
+		"app_c_with_a_very_long_name_abcdefghijklmnopqrstuvwxyz_0123456789/app_c_with_a_very_long_name_abcdefghijklmnopqrstuvwxyz_0123456789" : "app_c_with_a_very_long_name_abcdefghijklmnopqrstuvwxyz_0123456789",
+		"app_d_with_a_very_long_name_abcdefghijklmnopqrstuvwxyz_0123456789/app_d_with_a_very_long_name_abcdefghijklmnopqrstuvwxyz_0123456789" : "app_d_with_a_very_long_name_abcdefghijklmnopqrstuvwxyz_0123456789"
+	},
+	paths: {
+		"bundle.js": "bundle.js",
+		"steal/*" : "../../node_modules/steal/*.js",
+		"@traceur": "../../node_modules/traceur/bin/traceur.js"
+	},
+	bundle: [
```
Comment (justinbmeyer) [inline test/long_bundle_names/stealconfig.js]: Why does this file look like this?  A bundle should presumably not look like this.

File: test/long_bundle_names/bundle.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+steal(function(){
+	// Test: See if 'dep_all.js' is on the page
+	if (window.location.hash == '#a') {
+		steal('app_a_with_a_very_long_name_abcdefghijklmnopqrstuvwxyz_0123456789.js', function(appA){
```
Comment (justinbmeyer) [inline test/long_bundle_names/bundle.js]: this seems wrong too.  You shouldn't steal the bundle name.

Comment (justinbmeyer) [inline test/long_bundle_names/bundle.js]: nm, this is the module's name

File: test/long_bundle_names/stealconfig.js (hunk 1 above)
Comment (justinbmeyer) [inline test/long_bundle_names/stealconfig.js]: never mind. These are file names.

**Comment by alexisabril** [comment]
Closed by https://github.com/bitovi/steal-tools/pull/134

**Pull request closed without merging**